torch/onnx/experimental/fonnx/v1/exporter.py

```diff
@@ -0,0 +1,41 @@
+import io
+
+import torch
+
+from typing import Union
+
+
+def dynamo_export(
+    torch_model: torch.nn.Module,
+    *args,
+    opset_version: Optional[int] = None,
+    options: Optional[ExportOptions] = None,
+    verbose: Optional[bool] = None,
+    **kwargs,
+) -> ExportOutput:
+    """Export a torch.nn.Module to a ONNX graph
+
+    Args:
+        torch_model: User model to be exported
+        args: Positional inputs to `torch_model`
+        opset_version: Target ONNX opset version.
+        options: Advanced export options
+        verbose: Print basic debug when True
+        kwargs: Keyword inputs to `torch_model`
+
+    Returns:
+        An in-memory representation of the exported ONNX model.
+        See ``ExportOutput`` for reference.
+    """
+
+    if not options:
+        options = DefaultExportOptions(opset_version=opset_version, verbose=verbose)
+
+
+class ExportOutput:
+    def __init__(self, output: onnx.ModelProto):
+        ...
+
+    def save(self, destination: Union[str, io.BytesIO]) -> None:
```

Comment:
I don't have a strong preference. @abock ?

Reply:
I'd much prefer the union. Why does this need to return anything? For the sake of bikeshedding, I prefer `write` but in reality I would align with whatever the idiom is elsewhere in PyTorch. I agree that `destination` is an appropriate argument name for the union case.

Reply:
Because I hated the extra lines and calls needed for using io.BytesIO 🤣. I'm not aware of any use cases of `io.BytesIO` in context of `torch.onnx.export` that was not immediately getting the bytes result right after. But if you feel one call is much better than two, then we can keep it this way.

```python
f = io.BytesIO()
export_output.save(f)
model_bytes = f.getvalue()
```

vs

```python
model_bytes = export_output.save_to_bytes()
```

Reply:
Indeed, I also haven't seen `io.BytesIO() being used in any other context other than being used to `onnx.save(f)` or `onnx.load(f.getvalue())`.  In fact, I always forget if it should be `f.getvalue()`, `f.get_value()` or `f.getValue()`. Cutting `io.BytesIO` out of the way and leaving it as an implementation detail is handy

Reply:
Adding both `save` and `save_to_buffer` for now